### ISSUE TITLE
Upgrade all Build Tool to use Docker >= 24.x.x

### ISSUE THREAD

**Issue by markmandel** [googleforgames/agones#3530]
We now have an up to date client build of Docker on Cloud Build, if you look at the `24.0.6` tag on [gcr.io/cloud-builders/docker](https://gcr.io/cloud-builders/docker)

By default Cloud Build uses 20.x.x version, which has some quirks with manifest management, and usage of experimental docker CLI commands.

We should update all the Cloud Build scripts and Make tooling to be inline with Docker 24.x.x

in the Makefile, it's mainly this:

https://github.com/markmandel/agones/blob/9e7d596cc73a10eb6fc2f16d0097e0df14717d3d/examples/simple-game-server/Makefile#L90-L96

To switch over to use `docker buildx imagetools create` when creating manifests.

We should so check for `DOCKER_CLI_EXPERIMENTAL` as that may also want to be updated to a newer version of the commands.

At the same time, we will need to go through all references to `gcr.io/cloud-builders/docker` and point it specifically at `gcr.io/cloud-builders/docker:24.0.6`.

I'd recommend starting with the `/examples` images first, since that is low risk, and we can expand out from there.

**Comment by markmandel** [comment]
Coming off review of https://github.com/googleforgames/agones/pull/3531#issuecomment-1841561817 - @Kalaiselvi84 highlighted two things to note:

* `DOCKER_CLI_EXPERIMENTAL=enabled` should be removed from all commands, since it's no longer needed
* Where you see `docker buildx create --name=$(BUILDX_WINDOWS_BUILDER) --use` while not 100% necessary, I think it would be better to remove the `--use` so it's not set as the default for all future builds, and instead be explicit in our `docker buildx build` commands with the `--builder` argument.

Example of using `--build` can be found here:

https://github.com/googleforgames/agones/blob/330a548e24968bc0be7b2f2d9b5fc7e39787b86e/examples/xonotic/Makefile#L86-L88

Although in this example, we still need to remove `--use` and the `DOCKER_CLI_EXPERIMENTAL` parts of the commands.

Lemme know if any of that doesn't make sense.

**Comment by Kalaiselvi84** [comment]
Tried without `--use` and `DOCKER_CLI_EXPERIMENTAL=enabled`, but the build for simple-game-server failed:  https://gist.github.com/Kalaiselvi84/38a273a6dbb7e308fdb4be166fc597fd

**Comment by markmandel** [comment]
Can you push the changes, so I can see what your cloudbuild.yaml actually looks like -- hard to debug without it 😄

**Comment by Kalaiselvi84** [comment]
Committed the changes.👍🏻

Log without `DOCKER_CLI_EXPERIMENTAL=enabled` and with `--use`: https://gist.github.com/Kalaiselvi84/0d0b5fcc8de6fb9f31ac0d8588f40fda

**Comment by Kalaiselvi84** [comment]
@markmandel, should we focus only on `xonotic` and `simple-game-server`, or include all example images? If we include all, shall I submit a PR for the rest? I've updated the Docker image to version 24.0.6 in all the CloudBuild scripts. Almost all builds have succeeded, the build for c++ is still running.

**Comment by Kalaiselvi84** [comment]
Then let's just focus on `xonotic` and `simple-game-server` for now.

**Comment by Kalaiselvi84** [comment]
Both xonotic and simple-game-server PRs have been merged. May I know if there are any further tasks required for this issue?

**Comment by markmandel** [comment]
There is - we should also change everything in `/build` - but let's tackle that after the next release as it's a bigger chunk of work we'll want to coordinate together.

**Comment by markmandel** [comment]
Next steps for this 🎉:

* Update /cloudbuild.yaml, and fix what breaks (exactly as you did previously).
  * https://github.com/googleforgames/agones/blob/1e78d88a27ae7d427e43f073aab7d28289b7d850/cloudbuild.yaml#L30
  * https://github.com/googleforgames/agones/blob/1e78d88a27ae7d427e43f073aab7d28289b7d850/cloudbuild.yaml#L69

Before release, we'll also need to update the release process:
* https://github.com/googleforgames/agones/blob/1e78d88a27ae7d427e43f073aab7d28289b7d850/build/release/post_cloudbuild.yaml#L63

Once this is done, it might also be worth going through any other cloudbuild scripts looking for references to `gcr.io/cloud-builders/docker` and align everything on `gcr.io/cloud-builders/docker:24.0.6` to make everything consistent.

**Comment by markmandel** [comment]
Found another file that needs updating:

https://github.com/googleforgames/agones/blob/af751534ab82573f82d46f8328b3ba87fb3f8d07/ci/perf-test-cloudbuild.yaml#L26

https://github.com/googleforgames/agones/blob/af751534ab82573f82d46f8328b3ba87fb3f8d07/ci/perf-test-cloudbuild.yaml#L30

@gongmax - sorry looks like we broke your performance test CI.

**Comment by Kalaiselvi84** [comment]
@markmandel just double checking, is there anything pending in this Docker upgrade? or are we good to close this issue?

**Comment by markmandel** [comment]
Did a review! This is closeable!
